### ISSUE TITLE
Improve auto-generated TypeScript typing

### ISSUE THREAD

**Issue by DawChihLiou** [tantaraio/voy#2]
### Description 

Currently wasm-bindgen's `JsValue` translate to `any` type in TypeScript. The goal is to auto-generate correct typings for the Voy API.

### Tooling Candidates

- https://github.com/madonoharu/tsify
- https://lib.rs/crates/specta
- https://lib.rs/crates/typeshare
- https://lib.rs/crates/ts-rs

**Comment by jlarmstrongiv** [comment]
The current types are:
```ts
/* tslint:disable */
/* eslint-disable */
/**
* @param {any} input
* @returns {string}
*/
export function index(input: any): string;
/**
* @param {string} index
* @param {any} query
* @param {number} k
* @returns {any}
*/
export function search(index: string, query: any, k: number): any;
```

Are you looking for hand-crafted types or generated types?

I see you have more specific types in your README:

```ts
index(input: Resource): SerializedIndex

interface Resource {
  embeddings: Array<{
    id: string; // id of the resource
    title: string; // title of the resource
    url: string; // path to the resource
    embeddings: number[]; // embeddings of the resource
  }>;
}

type SerializedIndex = string; // serialized k-d tree
```

```ts
search(index: SerializedIndex, query: Query, k: NumberOfResult): Nearests

type SerializedIndex = string; // serialized k-d tree

type Query = number[]; // embeddings of the search query

type NumberOfResult = number; // K top results to return

type Nearests = Array<{
  id: string; // id of the nearest resource
  title: string; // title of the nearest resource
  url: string; // path of the nearest resource
  body: string; // body of the nearest resource
  embeddings: number[]; // embeddings of the nearest resource
}>;
```

**Comment by jlarmstrongiv** [comment]
Crates for generating typescript types include:
- https://lib.rs/crates/specta
- https://lib.rs/crates/typeshare
- https://lib.rs/crates/ts-rs

**Comment by DawChihLiou** [comment]
@jlarmstrongiv thanks for looking into it! I'm planning exploring [tsify](https://github.com/madonoharu/tsify) to generate the typings. Your suggestions look great. I'll look into them as well.

**Comment by DawChihLiou** [comment]
Released in v0.1.4🚀
